amr-wind/boundary_conditions/wall_models/WallFunction.cpp

```diff
@@ -116,14 +138,23 @@ void VelWallFunc::wall_model(
                         const amrex::Real vv =
                             vold_arr(i, j, k + idx_offset, 1);
                         const amrex::Real wspd = std::sqrt(uu * uu + vv * vv);
-                        // Dirichlet BC
-                        varr(i, j, k - 1, 2) = 0.0;
-
-                        // Shear stress BC
-                        varr(i, j, k - 1, 0) =
-                            tau.get_shear(uu, wspd) / mu * den(i, j, k);
+                        const amrex::Real xc = problo[0] + (i + 0.5) * dx[0]; 
+			const int z_dx_low = static_cast<int>(std::floor(dx[0] / dx[2])); 
```

Comment:
same comment here. It feels like there's an easier way...

Reply:
I'm trying to get the velocity field at a vertical height dx (z = dx). I do an interpolation because depending on the grid resolution it might be that velocity is not stored at that height.

Reply:
This makes me nervous too, and wouldn't `z_diff` be negative if `dx[0] = dx[2]` making `z_dx_low = 1`?
According to my hand calculation, this would lead to `z_diff = -0.5 * dx[2]`. 

Is the implicit assumption that `dx[2] < dx[0]`?

Reply:
Ok let's give all these some better names. `z_idx_lo` ? `z_idx_hi` and stuff like that?

Reply:
And if we are doing that, we need a check that k_lo/k_hi are in this box. `box.contains(iv)` should do it.